Add `is_deleted` column for funding and comment soft deletes

mddilley opened the pull request:
## Associated issues
https://github.com/cityofaustin/atd-data-tech/issues/9209

## Testing
**URL to test:** <!-- https://moped-test.austinmobility.io/moped/ or Netlify -->
_This PR needs to be tested either locally or on moped-test._

**Steps to test:**

**Comments**
1. Add a comment to a project through a status update in the Summary tab
2. Go to the project's Comments tab and add another comment through the WYSIWYG input
3. In the same tab, delete both comments you added in the Comment list view
4. You should see them disappear from the UI and remain soft deleted when you go to other tabs and return

**Funding**
1. Add a funding to a project on the Funding tab
2. In the same tab, delete the funding source that you just added
4. You should see it disappear from the UI and remain soft deleted when you go to other tabs and return

---
#### Ship list
- [x] Code reviewed 
- [x] Product manager approved
- [x] ~Added to [QA test script](https://docs.google.com/spreadsheets/d/1n_O6MLh9cwwPf57HUM394Ea-z9uuoEV1-QW4axNZXLE/edit#) if applicable~

chiaberry commented on moped-database/migrations/1653078890676_alter_table_public_moped_proj_notes_add_column_is_deleted/up.sql:
```diff
@@ -0,0 +1,9 @@
+ALTER TABLE "public"."moped_proj_notes" ADD COLUMN "is_deleted" boolean NOT NULL DEFAULT false;
+
+COMMENT ON COLUMN moped_proj_notes.is_deleted IS 'Indicates soft deletion';
+COMMENT ON COLUMN moped_proj_notes.status_id IS 'Deprecated soft deletion column';
```
i really like these comments and I am going to start using them ✨

frankhereford commented on moped-database/migrations/1654055457679_update_project_list_view_fund_notes_is_del/up.sql:
```diff
@@ -0,0 +1,97 @@
+CREATE OR REPLACE VIEW "public"."project_list_view" AS 
```
[Similar to the PR #656](https://github.com/cityofaustin/atd-moped/pull/656#discussion_r889060941), I would suggest a `drop view` statement before the `create`.

mddilley replied:
This is now covered in the latest migration pushed here. 🙏

frankhereford commented on moped-editor/src/views/projects/projectView/ProjectFundingTable.js:
```diff
@@ -662,6 +662,7 @@ const ProjectFundingTable = () => {
                   ...newData,
                   project_id: projectId,
                   added_by: getDatabaseId(user),
+                  // If no new funding status is selected, the default should be used
```
inline comments that explain intent -- awesome. 💯

frankhereford reviewed (approved):
![delete](https://user-images.githubusercontent.com/3653206/171901998-c21ed3cd-9bed-4248-b320-a603a410c732.gif)

Looks great @mddilley, thank you!

amenity commented:
FWIW, after deploying, the "Dashboard" page was taking a while to load so I switched to the "Projects" page and got this error:
<img width="787" alt="Screen Shot 2022-06-08 at 6 57 20 PM" src="https://user-images.githubusercontent.com/1463708/172736799-ef1844ae-7adb-4bdf-abf9-4d36d569c204.png">

Refreshed and everything loaded fine.

amenity reviewed (approved):
👍🏻 🗑️ 🙏🏻 🚢

mddilley commented:
Last commit contains changes from these steps:
1. Pulled in `main`
2. Resolved any conflicts
3. Incorporated changes to the `project_list_view` from my branch into version controlled SQL file
4. Updated view migration in PR to use that version of the `project_list_view`
5. Updated migrations introduced in this branch with fresh timestamps in sequence by milliseconds (to keep local working)

Happy to answer any and all questions about this.

mddilley commented:
Just a final summary here, my last commit updated the Dashboard queries for the **My Projects** and **Following** so they use `is_deleted` instead of `status_id` to show only comments that are not soft deleted.

mateoclarke reviewed (commented):
![lizzie-mcguire-delete](https://user-images.githubusercontent.com/5697474/174421946-27f30f8a-1d53-4b2a-9c80-8c0296dbb362.gif)
